lib/glueby/internal/wallet/active_record/key.rb

```diff
@@ -27,6 +27,38 @@ def address
             to_p2pkh.addresses.first
           end
 
+          # Return Glueby::Internal::Wallet::AR::Key object for input.
+          # If utxo spent by the specified input is colored output, key is found by corresponding `uncolored` script.
+          #
+          # @param [Tapyrus::TxIn] input
+          # @return [Glueby::Internal::Wallet::AR::Key] key for input
+          def self.key_for_input(input)
+            out_point = input.out_point
+            utxo = Utxo.find_by(txid: out_point.txid, index: out_point.index)
+            return unless utxo
+            script_pubkey = Tapyrus::Script.parse_from_payload(utxo.script_pubkey.htb)
```

Comment:
You can call `key_for_output` using this `script_pubkey`. As results, rest of following code can be removed.

Reply:
I added a new method Key.key_for_script (10abdae)